net-im/swift/swift-4.0.2.ebuild

```diff
@@ -0,0 +1,162 @@
+
+LICENSE="BSD BSD-1 CC-BY-3.0 GPL-3 OFL-1.1"
+SLOT="0/4"
+KEYWORDS="~amd64 ~x86"
+IUSE="expat examples experimental ft gconf hunspell lua icu idn qt5 test tools unbound zeroconf"
+REQUIRED_USE="ft? ( experimental )
+	gconf? ( qt5 )
+	hunspell? ( qt5 )"
+
+RDEPEND="dev-lang/lua:=
+	dev-libs/boost:=
+	dev-libs/openssl:0=
+	sys-libs/zlib:=
+	expat? ( dev-libs/expat )
+	!expat? ( dev-libs/libxml2:2 )
+	experimental? ( dev-db/sqlite:3 )
+	ft? (   net-libs/libnatpmp
+		net-libs/miniupnpc:= )
+	hunspell? ( app-text/hunspell:= )
+	icu? ( dev-libs/icu:= )
+	gconf? ( gnome-base/gconf:2 )
+	idn? ( net-dns/libidn:= )
+	qt5? ( dev-qt/qtcore:5
+		dev-qt/qtdbus:5
+		dev-qt/qtgui:5
+		dev-qt/qtmultimedia:5
+		dev-qt/qtnetwork:5
+		dev-qt/qtsvg:5
+		dev-qt/qtwidgets:5
+		dev-qt/qtwebkit:5
+		dev-qt/qtx11extras:5 )
+	unbound? ( net-dns/unbound
+		net-libs/ldns )
+	zeroconf? ( net-dns/avahi )"
+
+DEPEND="qt5? ( dev-qt/linguist-tools:5 )
+	>=dev-util/scons-3.0.1-r3
+	${RDEPEND}"
+
+DOCS=(
+	"DEVELOPMENT.md"
+	"README.md"
+	"Swiften/ChangeLog.md"
+)
+
+PATCHES=(
+	"${FILESDIR}"/${P}-make-generated-files-handle-unicode-characters.patch
+	"${FILESDIR}"/${P}-qt-5.11-compat.patch
+)
+
+src_prepare() {
+	default
+
+	# Hack for finding qt system libs
+	mkdir "${T}"/qt || die
+	ln -s "${EPREFIX%/}"/usr/$(get_libdir)/qt5/bin "${T}"/qt/bin || die
+	ln -s "${EPREFIX%/}"/usr/$(get_libdir)/qt5 "${T}"/qt/lib || die
+	ln -s "${EPREFIX%/}"/usr/include/qt5 "${T}"/qt/include || die
+
+	# Remove parts of Swift, which a user don't want to compile
+	! use lua && rm -fr Sluift
+	! use qt5 && rm -fr Swift Slimber
+	! use tools && rm -fr SwiftTools
+	! use zeroconf && rm -fr Limber
+
+	# Remove '3rdParty', as the system libs should be used
+	rm -fr 3rdParty || die
+}
+
+src_configure() {
+	MYSCONS=(
+		ar="$(tc-getAR)"
+		allow_warnings="yes"
+		assertions="no"
+		build_examples="yes"
+		boost_bundled_enable="false"
+		boost_force_bundled="false"
+		cc="$(tc-getCC)"
+		ccache="no"
+		ccflags="${CFLAGS}"
+		coverage="no"
+		cxx="$(tc-getCXX)"
+		cxxflags="${CXXFLAGS}"
+		debug="no"
+		distcc="no"
+		experimental="$(usex experimental)"
+		experimental_ft="$(usex ft)"
+		hunspell_enable="$(usex hunspell)"
+		icu="$(usex icu)"
+		install_git_hooks="no"
+		libidn_bundled_enable="false"
+		libminiupnpc_force_bundled="false"
+		libnatpmp_force_bundled="false"
+		link="$(tc-getCXX)"
+		linkflags="${LDFLAGS}"
+		max_jobs="no"
+		optimize="no"
+		qt="${T}/qt"
+		qt5="$(usex qt5)"
+		swiften_dll="true"
+		swift_mobile="no"
+		target="native"
+		test="none"
+		try_avahi="$(usex zeroconf)"
+		try_expat="$(usex expat)"
+		try_gconf="$(usex gconf)"
+		try_libidn="$(usex idn)"
+		try_libxml="$(usex !expat)"
+		tls_backend="openssl"
+		unbound="$(usex unbound)"
+		V="1"
+		valgrind="no"
+		zlib_bundled_enable="false"
+	)
+}
+
+src_compile() {
+	local MYSCONSTARGETS=(
+		Swiften
+		$(usex lua Sluift '')
+		$(usex qt5 Slimber '')
+		$(usex qt5 Swift '')
```

Comment:
Do I understand correctly that USE=-qt5 means no client is built? In that case, that's more like USE=client than USE=qt5. Or possibly USE=cli and USE=gui if you want to make them buildable separately.

Reply:
Thanks, fixed.

Reply:
This is correct. `USE="-qt5"` means, that only the library is build. USE={cli,gui} won't work, since the client always needs the lib. I've changed now `qt5` to `client`. But I still think, qt5 would be better, since most packages use this for a gui..

Reply:
Hmm, I'm confused. What do the 'Swift' and 'Slimber' targets respectively mean?

Reply:
`Swiften` = `The library itself`
`Swift` = `The XMPP client`
`Sluift` = `XMPP scripting environment based on Lua`
`SwifTools` = `Collection of utilities for Swift`
`Limber` = `XMPP proxy server proxying zero-configuration (XEP-0174) to a connected XMPP client`
`Slimber` = GUI for `Limber`